### PR TITLE
Allow IPv6 loopback address

### PR DESCRIPTION
**Pull request by Timer**
**What kind of change does this PR introduce?**

bugfix

**Did you add or update the `examples/`?**
N/A

**Summary**

localhost resolves to `::1` for IPv6, `127.0.0.1` for IPv4. We shouldn't forget about IPv6 users.

**Does this PR introduce a breaking change?**

No

**Other information**

### CODE REVIEW

**Comment by Timer** [comment]
Hi! I made these changes on github so I hope they work. 😅
In a nutshell, there was no support for IPv6... this remedies that!

File: lib/Server.js [hunk 1]
```diff
@@ -405,19 +406,17 @@ Server.prototype.checkHost = function(headers) {
...
 
 	// always allow localhost host, for convience
-	if(hostname === "127.0.0.1" || hostname === "localhost") return true;
+	if(hostname === "127.0.0.1" || hostname === "localhost" || hostname === "::1") return true;
 
 	// allow hostname of listening adress
 	if(hostname === this.listenHostname) return true;
 
 	// also allow public hostname if provided
 	if(typeof this.publicHost === "string") {
```
Comment (Timer) [inline lib/Server.js]: It would be great if `publicHost` could be an array so we could provide an IPv6 and IPv4 address.

**Comment by shellscape** [comment]
@Timer looks like there are some issues with the tests to resolve before we can merge this. Please have a look

File: lib/Server.js [hunk 2]
```diff
@@ -489,7 +485,7 @@ Server.prototype.listen = function(port, hostname) {
 	});
 	sockServer.on("connection", (conn) => {
 		if(!conn) return;
-		if(!this.checkHost(conn.headers)) {
+		if(!this.checkHost(conn.hostname)) {
```
Comment (Timer) [inline lib/Server.js]: We just have to make sure this works.

Comment (shellscape) [inline lib/Server.js]: it looks obvious as to why this was changed to `hostname` but what I'm wondering is if there was reason it was using `headers`. I think we may need tests around this to be certain.

**Comment by shellscape** [comment]
ping @Timer re: tests for headers/hostname change

**Comment by shellscape** [comment]
@Timer last ping before we close this one as abandoned. please do have a look at the conflicts and adding a test around the `hostname` change!

**Pull request closed without merging**